### ISSUE TITLE
MoE: aperture violation in gemv_hfq4g256_moe_gate_up_k8_indexed (Q8 KV path, 122B-A10B)

### ISSUE THREAD

**Issue by Kaden-Schutt** [Kaden-Schutt/hipfire#152]
## Symptom

`HSA_STATUS_ERROR_MEMORY_APERTURE_VIOLATION` during decode on Qwen3.5-122B-A10B (MQ4) when running with **Q8 KV cache** (the default for `infer_qwen35`). Crash is in the kernel \`gemv_hfq4g256_moe_gate_up_k8_indexed\`. Switching to **asym3 KV cache** runs to 2048 tokens cleanly with no crash.

## Repro

- **Hardware**: hipx (Minisforum MS-S1 MAX, Ryzen AI MAX+ 395 / Strix Halo, gfx1151, 96 GB iGPU UMA pool).
- **Branch**: master @ \`ba31a90\` (#146 merged) + cherry-pick of JIT shim trio \`8511ece + 4bb29a2 + 84a989f\` from \`strix-halo\`.
- **Model**: \`qwen3.5-122b-a10b.mq4\` (65 GB, freshly quantized via \`hipfire-quantize --format mq4\` from \`Qwen/Qwen3.5-122B-A10B\` HF safetensors).
- **Run**:
  \`\`\`
  HIP_VISIBLE_DEVICES=0 ./target/release/examples/infer_qwen35 \\
    ~/.hipfire/models/qwen3.5-122b-a10b.mq4 "<some long prompt>"
  \`\`\`
- **Default KV mode** (Q8 — no \`HIPFIRE_KV_MODE\` set): crashes after ~500-1000 decoded tokens (during deep degenerate-output phase).
- **Same run with \`HIPFIRE_KV_MODE=asym3\`**: completes 2048 tokens at 36.2 tok/s, no crash.

## Crash signature

\`\`\`
:0:rocdevice.cpp            :3586: ... us:  Callback: Queue 0x... aborting with error :
HSA_STATUS_ERROR_MEMORY_APERTURE_VIOLATION: The agent attempted to access memory beyond
the largest legal address. code: 0x29

thread 'main' panicked at crates/engine/examples/infer_qwen35.rs:230:
called \`Result::unwrap()\` on an \`Err\` value:
HipError { code: 700, message: "hipMemcpy D2H: an illegal memory access was encountered" }

Kernel Name: gemv_hfq4g256_moe_gate_up_k8_indexed
VGPU=..., SWq=..., HWq=...
Dispatch Header =0xb02 (type=2, barrier=1, acquire=1, release=1), setup=0
grid=[65536, 8, 1], workgroup=[32, 1, 1]
private_seg_size=0, group_seg_size=0
\`\`\`

## Hypothesis

Two candidates worth investigating:

1. **Routing-index out-of-bounds**: by the time the crash hits, output is already in attractor degeneration (#145 quality bug). Degenerate logits may produce extreme routing values that select invalid expert IDs. The \`_indexed\` kernel reads a routing tensor produced by the previous layer's output and may not bounds-check.

2. **Q8-path-specific buffer alignment**: \`asym3\` KV doesn't trigger the same crash with the same prompt. KV mode shouldn't directly affect MoE GEMV, but Q8-vs-asym3 produces different memory layouts that may interact with the MoE gate_up kernel's expected stride/alignment when 122B-A10B's specific dimensions hit.

## Why this matters

122B-A10B is the largest MoE Qwen3.5 hipfire targets. The Q8 KV crash blocks any work that needs the legacy KV path on this model. \`asym3\` works as a workaround, but a kernel that crashes silently on degenerate input is a latent bug — could fire on smaller models with similar attractor patterns.

Refs: #145 (parent quality bug — same model, MQ4 expert quant error)
